ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -347,6 +347,41 @@ let dss_loadavg () =
     )
   ]
 
+let dss_hostload xc domains =
+  let physinfo = Xenctrl.physinfo xc in
+  let pcpus = physinfo.Xenctrl.nr_cpus in
+  let load =
+    List.fold_left
+      (fun acc (dom, _, domid) ->
+        let vcpu_ids = List.init (dom.Xenctrl.max_vcpu_id + 1) (fun i -> i) in
```

Comment:
This is two nested `List.fold_left` and hence quadratic effort. Is this the best we can do? Also, it appears that `vcpu_ids` is simply to iterate over a list of numbers - we don't have to turn this into a list but could use a loop that increments a counter. Code like this should create less memory garbage.

Reply:
No exiting interface to get all vGPU info directly, so have to get from domains indirectly. 
I have optimized to in loop way, could you help to review further?